### DISCUSSION TITLE
Preferences: Disable Autoformatting

### DISCUSSION THREAD

**Discussion by dblegend** [zim-desktop-wiki/zim-desktop-wiki#2196] [Ideas]
I've been using Zim for at least 8 years in Linux, Windows, and MacOS, and have found it to be an amazing tool for a number of use cases and have recommended it to many people! Something that has always been a frustration to me, however, is the auto formatting. I'd very much like an option to disable it in some permanent capacity. Despite scouring the documentation several times, I haven't found a good way to accomplish this.

For example, in my notes, I often use a long string of hyphens (e.g. "-----------------------------------") as a section break. Zim auto formats that to a full, colored, double bar line break. (A few years ago, this finally drove me so crazy that I gave up and switched to using a string of periods.) Anytime I include an email address in the text, it gets auto formatted to a clickable link. Even when I remove that link, it just returns when the page is reloaded.

Recently, I've been learning Python while keeping syntax notes in a Zim notebook. Anytime I paste in a code block containing a function header (e.g. _ _ function _ _), that gets auto-formatted (removing the underscores, and highlighting the content) as opposed to just leaving the text. Jinja2 (python / Flask) syntax is full of "{{ }}" type blocks, and those are getting auto formatted in undesirable ways as well. Worse still, after the formatting is complete, the original text isn't displayed, which significantly impedes syntax review / acquisition. (To get around this, I'm having to resort to a source code plugin, which works, but adds additional clicks, gets in the way of the work, and is just generally cumbersome for my needs.)

Possibly more frustrating is that, even when I ctrl+z the auto formatting to set it back to plain text, Zim has never remembered the preference. Reloading the page, or closing / reopening the software, causes all of it to get reapplied to the page all over again! I have yet to find a way to remove formatting from a line of text or even on an entire page and actually get it to stick.

My feature request is to offer a user control to disable auto-formatting entirely - either at a per-page, or per-notebook level. Even better, I'd love it if Zim had a way to actually remember when I opt to undo / clear formatting and for it to persist across reloads and restarts. I would even be excited about a simple markup block that, when read by the formatter, told it "Anything in this block needs to be left alone!"

Thanks again for all the great work, and for considering this idea!

**Comment by introt** [comment]
Most of what you're struggling with isn't auto-formatting per se, but the flip side of Zim's combination of wiki syntax & What-You-See-Is-What-You-Get -editing (see eg. https://en.wikipedia.org/wiki/Wiki_format#Basic_syntax).

> I have yet to find a way to remove formatting from a line of text or even on an entire page and actually get it to stick.

> I would even be excited about a simple markup block that, when read by the formatter, told it "Anything in this block needs to be left alone!"

That'd be the `Verbatim` format - available as both inline as well as block variants. As with the horizontal divider bar, you can modify its display style to your taste using `style.conf` - see the manual for options.

Once we have #26 in the bag, we'll likely have the necessary infrastructure in place to support "raw" pages etc.

See also Escape sequences for verbatim content #577

**Reply by dblegend** [reply]
Verbatim text blocks are exactly what I have been needing. I'm not sure how I missed this right there in the editor GUI all these years, but I'll be using these blocks regularly in the future. Thank you!
